temp.c

```diff
@@ -173,65 +259,22 @@ void temp_sensor_tick() {
 
 				#ifdef	TEMP_THERMISTOR
 				case TT_THERMISTOR:
-					do {
-						uint8_t j, table_num;
-						//Read current temperature
-						temp = analog_read(i);
```

Comment:
And then there is this thread: http://forums.reprap.org/read.php?1,647767,648074#msg-648074 Is this the reason why the existing code was suspect to you?

Other than that: nice patch! I'll apply it before too long.